lightning/src/ln/channel.rs

```diff
@@ -1684,63 +1735,172 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
+				next_holder_htlc_id: match htlc.origin {
+					HTLCInitiator::LocalOffered => self.next_holder_htlc_id + 1,
+					HTLCInitiator::RemoteOffered => self.next_holder_htlc_id,
+				},
+				next_counterparty_htlc_id: match htlc.origin {
+					HTLCInitiator::LocalOffered => self.next_counterparty_htlc_id,
+					HTLCInitiator::RemoteOffered => self.next_counterparty_htlc_id + 1,
+				},
+				feerate: self.feerate_per_kw,
+			};
+			*self.next_local_commitment_tx_fee_info_cached.lock().unwrap() = Some(commitment_tx_info);
+		}
+		res
 	}
 
-	// Get the commitment tx fee for the remote's next commitment transaction
-	// based on the number of pending HTLCs that are on track to be in their
-	// next commitment tx. `addl_htcs` is an optional parameter allowing the caller
-	// to add a number of additional HTLCs to the calculation. Note that dust HTLCs
-	// are excluded.
-	fn next_remote_commit_tx_fee_msat(&self, addl_htlcs: usize) -> u64 {
+	// Get the commitment tx fee for the remote's next commitment transaction based on the number of
+	// pending HTLCs that are on track to be in their next commitment tx, plus an additional HTLC if
+	// `fee_spike_buffer_htlc` is Some, plus a new HTLC given by `new_htlc_amount`. Dust HTLCs are
+	// excluded.
+	fn next_remote_commit_tx_fee_msat(&self, htlc: HTLCCandidate, fee_spike_buffer_htlc: Option<()>) -> u64 {
 		assert!(!self.is_outbound());
 
-		// When calculating the set of HTLCs which will be included in their next
-		// commitment_signed, all inbound HTLCs are included (as all states imply it will be
-		// included) and only committed outbound HTLCs, see below.
-		let mut their_acked_htlcs = self.pending_inbound_htlcs.len();
+		let real_dust_limit_success_sat = (self.feerate_per_kw as u64 * HTLC_SUCCESS_TX_WEIGHT / 1000) + self.counterparty_dust_limit_satoshis;
+		let real_dust_limit_timeout_sat = (self.feerate_per_kw as u64 * HTLC_TIMEOUT_TX_WEIGHT / 1000) + self.counterparty_dust_limit_satoshis;
+
+		let mut addl_htlcs = 0;
+		if fee_spike_buffer_htlc.is_some() { addl_htlcs += 1; }
+		match htlc.origin {
+			HTLCInitiator::LocalOffered => {
+				if htlc.amount_msat / 1000 >= real_dust_limit_success_sat {
+					addl_htlcs += 1;
+				}
+			},
+			HTLCInitiator::RemoteOffered => {
+				if htlc.amount_msat / 1000 >= real_dust_limit_timeout_sat {
+					addl_htlcs += 1;
+				}
+			}
+		}
+
+		// When calculating the set of HTLCs which will be included in their next commitment_signed, all
+		// non-dust inbound HTLCs are included (as all states imply it will be included) and only
+		// committed outbound HTLCs, see below.
+		let mut included_htlcs = 0;
+		for ref htlc in self.pending_inbound_htlcs.iter() {
+			if htlc.amount_msat / 1000 <= real_dust_limit_timeout_sat {
+				continue
+			}
+			included_htlcs += 1;
+		}
+
 		for ref htlc in self.pending_outbound_htlcs.iter() {
-			if htlc.amount_msat / 1000 <= self.counterparty_dust_limit_satoshis {
+			if htlc.amount_msat / 1000 <= real_dust_limit_success_sat {
```

Comment:
I think it would be great to comment this subtlety around which dust limit apply in function of which viewpoint the fee is computed for.

"When computing the fee as paid by counterparty, if they have to claim onchain HTLC by broadcasting their commitment, they will have to spend inbound through a HTLC-timeout, outbound through a HTLC-Success. 
When computing the fee as paid by holder, if we have to claim onchain HTLC by broadcasting our commitment, we will have to spend inbound through a HTLC-success, outbound through a HTLC-timeout"

Well, that's an attempt, but we should make it clear that even if our channel variables represent the holder direction-side (`pending_inbound_htlcs`/`pending_outbound_htlcs`) the actual fee cost is function from which commitment they will be claim.